### PR TITLE
Use iComfort client code from module

### PR DESCRIPTION
**Pull request by danielmcq**
I'm happy to see a real world use of my iComfort client in action. When I realized that it had been used in this project, I updated the client code to get it ready for publishing in npm as a module. This pull request is for an update to the alexa-icomfort project to have it use the [icomfort](https://github.com/deHugo/icomfort-js) module as a dependency.

### CODE REVIEW

**Comment by kate-hall** [comment]
@danielmcq ahh!! This is great, thank you so much (and thanks for developing the client in the first place). I'll test this branch on my thermostat tonight before merging.

**Comment by kate-hall** [comment]
@danielmcq Annoyingly, AWS Lambda doesn't yet support all of the ES6 functionality, particularly some you're using in the module (http://whatdoeslambdasupport.com/). Right now I'm getting this error:
screen shot 2017-02-14 at 3 58 10 pm

This one's related to the lack of support for default parameters.

 **EDIT** Ignore! Changing just that one line lets it run again! I'll make the change in my local module.

**Comment by danielmcq** [comment]
Good to know about AWS Lambda. I wasn't aware of the limitation.

Shall we close this pull request?

**Comment by kate-hall** [comment]
Nope, let's leave it open until I can find a way to make it work, I like the idea of moving to your module, just investigating options!

**Comment by danielmcq** [comment]
@kate-hall I've updated the icomfort module to version 1.1.1. I made sure to test it with Node 4.3.2, which is the version which [AWS Lambda uses](http://docs.aws.amazon.com/lambda/latest/dg/current-supported-versions.html). I updated this pull request to reflect that change. I believe this will correctly work for you now.

**Comment by kate-hall** [comment]
Awesome! I'll test this evening.

File: src/index.js [hunk 1]
```diff
@@ -132,7 +133,7 @@ function handleControl(event, context) {
             }
 
             // send the change request to Lennox, send a response to Alexa on promise fulfillment
-            iComfort.setThermostatInfo(getThermostatInfoListParams, newParams.toSet, auth)
+            iComfort.setThermostatInfo(getThermostatInfoListParams, newParams.toSet)
```
Comment (kate-hall) [inline src/index.js:136]: @danielmcq I believe this line should be `iComfort.setThermostatInfo(newParams.toSet)` for compatibility with the node module, if you want to adjust?

File: tests/index.js [hunk 2]
```diff
@@ -30,21 +32,21 @@ var getSystemsInfoParams = {UserId: auth.username};
...
 
 /* Test 3 = Returns data on min and max temperatures of your unit, not used in Alexa Skill */
-iComfort.getGatewayInfo(getInfoParams, auth)
+iComfort.getGatewayInfo(getInfoParams)
     .then(console.log)
     .catch(console.error);
 
 /* Test 4 = Careful! This test will change your actual temperature, uncomment to run */
-// iComfort.setThermostatInfo(getInfoParams, changeTemperatureTestVars, auth)
+// iComfort.setThermostatInfo(getInfoParams, changeTemperatureTestVars)
```
Comment (kate-hall) [inline tests/index.js:50]: @danielmcq Ditto the above, this line should be `iComfort.setThermostatInfo(changeTemperatureTestVars)` for compatibility with the node module, if you want to adjust?

**Pull request merged**